explorer/src/controllers/admin/nodes.ts

```diff
@@ -0,0 +1,51 @@
+import { Router, Request, Response } from 'express'
+import { Connection } from 'typeorm'
+import { validate } from 'class-validator'
+import { getDb } from '../../database'
+import { buildChainlinkNode, ChainlinkNode } from '../../entity/ChainlinkNode'
+import { PG_UNIQUE_CONSTRAINT_VIOLATION } from '../../utils/constants'
+
+const router = Router()
+
+router.post('/nodes', async (req: Request, res: Response) => {
+  const name: string = req.body.name
+  const url: string = req.body.url
+  const db: Connection = await getDb()
+  const [node, secret] = await buildChainlinkNode(db, name, url)
+  const errors = await validate(node)
+
+  if (errors.length === 0) {
+    try {
+      const savedNode = await db.manager.save(node)
+
+      return res.status(201).json({
+        id: savedNode.id,
+        accessKey: savedNode.accessKey,
+        secret: secret,
+      })
+    } catch (e) {
+      if (e.code === PG_UNIQUE_CONSTRAINT_VIOLATION) {
```

Comment:
e is going to have the type of `any` here. we might want a typecast or typeguard so we dont have any `any` types leaking through

Reply:
How do you recommend it should be typecast as the catch variable can't be annotated with a type. Are you thinking to just create a new variable? e.g.

```typescript
const caughtError = <SomeDatabaseError>e
```

Reply:
Also how do you recommend typing this? It returns a `QueryFailedError` but it appears that the driver attributes are dynamically assigned and not typed. 

https://github.com/typeorm/typeorm/blob/master/src/error/QueryFailedError.ts